cachi2/core/package_managers/pip.py

```diff
@@ -779,7 +779,7 @@ class PipRequirementsFile:
     # options apply to all the requirements.
     REQUIREMENT_OPTIONS = {"-e", "--editable", "--hash"}
 
-    def __init__(self, file_path):
+    def __init__(self, file_path: Union[Path, RootedPath, str]) -> None:
```

Comment:
```suggestion
    def __init__(self, file_path: Union[str, PathLike[str]]) -> None:
```
would be more general

Side note: this is what most stdlib methods accept as file paths these days